### ISSUE TITLE
No longer any way to always start new instance

### ISSUE THREAD

**Issue by smithfred** [gnome-mpv/gnome-mpv#321]
tl;dr: please restore the option to *always* use a new gnome-mpv process without connecting to the existing one.

As of 0.13 (I think), it's no longer possible to rely on a `gnome-mpv somefile` command only completing when the user closes the window showing the file.

It seems that gnome-mpv now *always* attempts to connect to an existing instance, with the command exiting immediately if one exists. This is the case with "Always open new window" as well.

This really breaks things for scripting, since it's not possible to rely on "process completed" meaning "video window has been closed".

A couple of use-cases:
* using gnome-mpv as a player for a temporarily downloaded video (e.g. `'youtube-dl --exec '\''gnome-mpv 2>/dev/null {} && rm {}'\'''`. With previous releases, after the gnome-mpv was closed, the file would be deleted. Now, if gnome-mpv is already open, `gnome-mpv` exits immediately, and the file gets deleted before gnome-mpv can even open it.
* a script that opens videos in gnome-mpv sequentially, playing the next one when the previous window was closed. If gnome-mpv isn't already running, this works fine, but if there's already a gnome-mpv window open, this can result in hundreds of gnome-mpv windows spawning in rapid succession.

**Comment by gnome-mpv** [comment]
For scripting, you can use `--no-existing-session` to restore that behavior. The dconf option was removed because it interacts badly with DBus activation (#312).

**Comment by TingPing** [comment]
It sounds like a better option is a `--wait` flag or such similar to what `gedit` has so it doesn't exit until the parent process does.

**Comment by smithfred** [comment]
@gnome-mpv thanks. That option's missing from the man page ;) So is --display actually. I've changed the bug title accordingly.

TingPing: a --wait switch would need to wait for the window associated with the specific file to be closed, rather than the entire single gnome-mpv process, to provide similar behaviour.

**Comment by TingPing** [comment]
> TingPing: a --wait switch would need to wait for the window associated with the specific file to be closed, rather than the entire single gnome-mpv process, to provide similar behaviour.

That is exactly how `--wait` on `gedit` behaves.

**Comment by TingPing** [comment]
> So is --display actually

`--display` is a Gtk option that nobody should use, it is removed in Gtk4. Use `DISPLAY` env var if you need to which is probably documented somewhere deep in the `xorg` manpages.

**Comment by gnome-mpv** [comment]
> @gnome-mpv thanks. That option's missing from the man page ;) So is --display actually. I've changed the bug title accordingly.

I'd prefer to leave it out as it is intended to be used only for debugging. I'll implement `--wait` later to handle your use case.

**Comment by smithfred** [comment]
Having a --wait option would be great as well, thanks for that 👍

Can I suggest adding the existing switch to man and explicitly marking it as "intended for debugging only, might eat your cat" then?

* for consistency with `--help`, and since it exists ;)
* having an "always use new process" option might still be handy in other cases as well (i.e. so it doesn't get removed when --wait is implemented) - e.g. one pathological vid won't take down every other open one; it's much easier to kill off individual vids from the process list, etc.

**Comment by gnome-mpv** [comment]
> for consistency with --help, and since it exists ;)

It's hidden from the `--help` output now: c0829ae51c2ac284873ed56d408caf4dad25f885

> having an "always use new process" option might still be handy in other cases as well (i.e. so it doesn't get removed when --wait is implemented) - e.g. one pathological vid won't take down every other open one; it's much easier to kill off individual vids from the process list, etc.

I have no plan to remove `--no-existing-session`.

**Comment by smithfred** [comment]
OK, I understand your thinking about wanting to hide it since it causes problems in some circumstances, though I still think explicitly warning about it and providing some sort of way of discovering it would be better for those who know what they're doing (or know the caveats).

I guess a more general fix would be the switch/option make it play nice with D-Bus since it covers use-cases that aren't entirely covered by "always new window" (even with `--wait`).

Thanks for the responsiveness and help anyway 😃
